### ISSUE TITLE
Image tag with data:image/svg not implemented

### ISSUE THREAD

**Issue by Cian-Chambliss** [vvvv/SVG#220]
for image tags of the form

<image xlink:href="data:image/svg+xml,..>

and

<image xlink:href="data:image/svg+xml;base64,..>

an exception is thrown by image.FromString(), which is expecting a bitmap image, not vector graphics.

Fixed this by changing the GetImage() method in SvgImage.cs,  first test the mime type portion of the message for image/svg, then get the SVG string & create an SVG Document.

```
    protected object GetImage(Uri uri)
    {
        try
        {
            // handle data/uri embedded images (http://en.wikipedia.org/wiki/Data_URI_scheme)
            if (uri.IsAbsoluteUri && uri.Scheme == "data")
            {
                string uriString = uri.OriginalString;
                int dataIdx = uriString.IndexOf(",") + 1;
                if (dataIdx <= 0 || dataIdx + 1 > uriString.Length)
                    throw new Exception("Invalid data URI");
                string imageType = uriString.Substring(0, dataIdx); // Check for SVG embedded in the data...
                if (imageType.Contains("image/svg"))
                    {
                        string embeddedSVG;
                        if (imageType.Contains("base64") )
                        {
                            embeddedSVG = System.Text.Encoding.UTF8.GetString( Convert.FromBase64String(uriString.Substring(dataIdx)) );
                        }
                        else
                        {
                            embeddedSVG = uriString.Substring(dataIdx);
                        }
                        using (var stream = new MemoryStream(System.Text.UTF8Encoding.Default.GetBytes(embeddedSVG)))
                        {
                            var doc = SvgDocument.Open<SvgDocument>(stream);
                            doc.BaseUri = uri;
                            return doc;
                        }
                    }
                else
                    {
                        // we're assuming base64, as ascii encoding would be *highly* unsusual for images
                        // also assuming it's png or jpeg mimetype
                        byte[] imageBytes = Convert.FromBase64String(uriString.Substring(dataIdx));
                        using (var stream = new MemoryStream(imageBytes))
                        {
                            return Image.FromStream(stream);
                        }
                    }
            }

            if (!uri.IsAbsoluteUri)
            {
                uri = new Uri(OwnerDocument.BaseUri, uri);
            }

            // should work with http: and file: protocol urls
            var httpRequest = WebRequest.Create(uri);

            using (WebResponse webResponse = httpRequest.GetResponse())
            {
                using (var stream = webResponse.GetResponseStream())
                {
                    if (stream.CanSeek)
                    {
                        stream.Position = 0;    
                    }
                    if (uri.LocalPath.EndsWith(".svg", StringComparison.InvariantCultureIgnoreCase))
                    {
                        var doc = SvgDocument.Open<SvgDocument>(stream);
                        doc.BaseUri = uri;
                        return doc;
                    }
                    else
                    {
                        return Bitmap.FromStream(stream);
                    }
                }
            }
        }
        catch (Exception ex)
        {
            Trace.TraceError("Error loading image: '{0}', error: {1} ", uri, ex.Message);
            return null;
        }
    }
```

**Comment by mrbean-bremen** [comment]
@Cian-Chambliss - looks like this is still not supported. Do you have an example image for this? Maybe even make a PR from your changes if you are still interested?

**Comment by mrbean-bremen** [comment]
Fixed by #462.
